mlir/lib/Gradient/Transforms/GradMethods/ParameterShift.hpp

```diff
@@ -21,7 +21,7 @@
 
 #include "Gradient/IR/GradientOps.h"
 
-constexpr double PI = 3.14159265358979323846;
+constexpr double PI = llvm::numbers::pi;
```

Comment:
I noticed we don't actually use `PI` within this header, maybe it would make more sense to keep its definition local to where it's used.

As far as I can tell, it's only used in [`computePartialDerivative()`](https://github.com/PennyLaneAI/catalyst/blob/main/mlir/lib/Gradient/Transforms/GradMethods/PS_QuantumGradient.cpp#L59) at the moment, so we could probably just as easily write,

```cpp
constexpr double shift = llvm::numbers::pi / 2;
```

Reply:
I'm ok with either, since the `PI` was already in this header.

Reply:
I'm fine with either as well, just a nitpick :slightly_smiling_face: